src/librustc/ty/sty.rs

```diff
@@ -914,16 +944,16 @@ impl<'a, 'tcx, 'gcx> ExistentialProjection<'tcx> {
         ty::ProjectionPredicate {
             projection_ty: ty::ProjectionTy::from_ref_and_name(
                 tcx,
-                self.trait_ref.with_self_ty(tcx, self_ty),
-                self.item_name),
+                self.trait_ref(tcx).with_self_ty(tcx, self_ty),
+                self.item_name(tcx)),
             ty: self.ty,
         }
     }
 }
 
 impl<'a, 'tcx, 'gcx> PolyExistentialProjection<'tcx> {
-    pub fn item_name(&self) -> Name {
-        self.skip_binder().item_name()
+    pub fn item_name(&self, tcx: TyCtxt) -> Name {
+        self.skip_binder().item_name(tcx)
```

Comment:
Grepped the diff and there's one call added call to `item_name()` (but many removed). I inlined `from_ty_ref_and_name` at its only caller (which has the one call to `item_name()`):

```rust
        let existential_projections = projection_bounds.iter().map(|bound| {
            bound.map_bound(|b| {
                let trait_ref = self.trait_ref_to_existential(b.projection_ty.trait_ref(tcx));
                let item_name = b.projection_ty.item_name(tcx);
                let item_def_id = tcx.associated_items(trait_ref.def_id).find(
                    |item| item.name == item_name
                ).unwrap().def_id;

                ty::ExistentialProjection {
                    ty: b.ty,
                    item_def_id: item_def_id,
                    substs: trait_ref.substs,
                }
            })
        });
```

Maybe I can simplify the `item_name()` call away, but someone would have to help me out.